### ISSUE TITLE
Feature Request: Support for GetFeatureInfo on a WMTS layer using template from WMTS capabilities

### ISSUE THREAD

**Issue by mbosecke** [openlayers/openlayers#14582]
I'm using Geoserver as a backend to serve WMTS layers, which are built on the frontend using "optionsFromCapabilities". Within the capabilities document, I see URL templates on how to implement GetFeatureInfo that look like this:

`<ResourceURL format="text/xml" resourceType="FeatureInfo" template="https://mitchell.internal.forcorp.com/geoserver/gwc/service/wmts/rest/public:fma_simplified/{style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}/{J}/{I}?format=text/xml"/>
<ResourceURL format="text/html" resourceType="FeatureInfo" template="https://mitchell.internal.forcorp.com/geoserver/gwc/service/wmts/rest/public:fma_simplified/{style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}/{J}/{I}?format=text/html"/>
<ResourceURL format="application/json" resourceType="FeatureInfo" template="https://mitchell.internal.forcorp.com/geoserver/gwc/service/wmts/rest/public:fma_simplified/{style}/{TileMatrixSet}/{TileMatrix}/{TileRow}/{TileCol}/{J}/{I}?format=application/json"/>`

I would like a feature that helps build GetFeatureInfo URLs, using the templates provided from a WMTS capabilities document.

It looks like this pull request was on the right track but never completed: https://github.com/openlayers/openlayers/pull/3150

**Comment by mbosecke** [comment]
I've added a small bounty to this feature request with a 1 month expiration in case there's anyone who can help: 

https://app.bountysource.com/issues/118117964-feature-request-support-for-getfeatureinfo-on-a-wmts-layer-using-template-from-wmts-capabilities

**Comment by mike-000** [comment]
I don't think this would be a method of the WMTS source as it requires information not available to the source.  It is however easy to deduce the url needed for an info request given the parsed capabilities, the options used to create the source and the format required for the info, together with the coordinate and resolution, in the application using existing API methods. For example replacing the WMS with WMTS in the GeoFeatureInfo example https://codesandbox.io/s/getfeatureinfo-tile-forked-4418ms?file=/main.js
